### ISSUE TITLE
AppComponentsFactory | buildConstructorArgs() needs to accept an App instance as optional second argument

### ISSUE THREAD

**Issue by sevidmusic** [sevidmusic/DarlingDataManagementSystem#131]
AppComponentsFactory | buildConstructorArgs() needs to accept an App instance as an optional second argument. 

This will both allow the configuration of the App instance used by the AppComponentsFactory to be specified when appropriate, and make it possible to use alternative App implementations with an AppComponentsFactory.

This refactor of the AppComponentsFactory was motivated by the changes made to resolve issue #128. This issue and issue #128 both are related to making it possible to set the name of an App to something other than the alphanumeric camel case form of the App's domain.

**Comment by sevidmusic** [comment]
Original:
```
 public static function buildConstructorArgs(RequestInterface $domain): array 
    {   
        return [
            self::buildPrimaryFactory($domain),
            self::buildComponentCrud($domain),
            self::buildStoredComponentRegistry($domain)
        ];
    }   
```
Change:
``` public static function buildConstructorArgs(RequestInterface $domain, string $appName = ''): array 
    {   
        return [
            (isset($appName) ? $appName : self::buildPrimaryFactory($domain)),
            self::buildComponentCrud($domain),
            self::buildStoredComponentRegistry($domain)
        ];
    }   
```

**Comment by sevidmusic** [comment]
Resolved in commit 7edc18c0652a7a9d06f95ac445aa61a6e85da8b1
